timeseries.py

```diff
@@ -0,0 +1,650 @@
+
+    def append(self, other: 'TimeSeries') -> 'TimeSeries':
+        """
+        Appends another TimeSeries to this TimeSeries.
+
+        :param other: As second TimeSeries.
+        :return: A new TimeSeries, obtained by appending the second TimeSeries to the first.
+        """
+
+        assert other.start_time() == self.end_time() + self.freq(), 'Appended TimeSeries must start one time step ' \
+                                                                    'after current one.'
+
+        series = self._series.append(other.pd_series())
+        conf_lo = None
+        conf_hi = None
+        if self._confidence_lo is not None and other.conf_lo_pd_series() is not None:
+            conf_lo = self._confidence_lo.append(other.conf_lo_pd_series())
+        if self._confidence_hi is not None and other.conf_hi_pd_series() is not None:
+            conf_hi = self._confidence_hi.append(other.conf_hi_pd_series())
+        return TimeSeries(series, conf_lo, conf_hi)
+
+    @staticmethod
+    def _combine_or_none(series_a: Optional[pd.Series],
+                         series_b: Optional[pd.Series],
+                         combine_fn: Callable[[pd.Series, pd.Series], Any]):
+        """
+        Combines two Pandas Series [series_a] and [series_b] using [combine_fn] if neither is None.
+
+        :param series_a: A Pandas Series.
+        :param series_b: A Pandas Series.
+        :param combine_fn: An operation with input two Pandas Series and output one Pandas Series.
+        :return: A new Pandas Series, the result of [combine_fn], or None.
+        """
+
+        if series_a is not None and series_b is not None:
+            return combine_fn(series_a, series_b)
+        return None
+
+    @staticmethod
+    def _op_or_none(series: Optional[pd.Series], op: Callable[[pd.Series], Any]):
+        return op(series) if series is not None else None
+
+    def _combine_from_pd_ops(self, other: 'TimeSeries',
+                             combine_fn: Callable[[pd.Series, pd.Series], pd.Series]) -> 'TimeSeries':
+        """
+        Combines this TimeSeries with another one, using the [combine_fn] on the underlying Pandas Series.
+
+        :param other: A second TimeSeries.
+        :param combine_fn: An operation with input two Pandas Series and output one Pandas Series.
+        :return: A new TimeSeries, with underlying Pandas Series the series obtained with [combine_fn].
+        """
+
+        assert self.has_same_time_as(other), 'The two TimeSeries must have the same time index.'
+
+        series = combine_fn(self._series, other.pd_series())
+        conf_lo = self._combine_or_none(self._confidence_lo, other.conf_lo_pd_series(), combine_fn)
+        conf_hi = self._combine_or_none(self._confidence_hi, other.conf_hi_pd_series(), combine_fn)
+        return TimeSeries(series, conf_lo, conf_hi)
+
+    """
+    Definition of some methods related to seasonality.
+    """
+
+    def check_seasonality(self, m: int = 0) -> Tuple[bool, int]:
```

Comment:
How about moving these seasonality methods to a different place, e.g. in `u8timeseries/models/seasonality/...` ?

This way we don't overload the `TimeSeries` object with models, and keep the models in separate "modules"?

Reply:
Yes it would be better. I would maybe go for a "statistics" which also contains tests for trends and stationarity. To be discussed.